### PR TITLE
allow searching through friends in invites page

### PR DESCRIPTION
**Pull request by gmethvin**

### CODE REVIEW

File: kifi-backend/modules/common/public/site/js/main.js [hunk 1]
```diff
@@ -566,29 +566,30 @@ $(function() {
...
 	var $nwFriendsLoading = $('.invite-friends-loading');
-	function prepInviteTab() {
-		$.getJSON(xhrBase + '/user/socialConnections', function(friends) {
+	var friendsToShow = 20;
+	var friendsTimeout;
+	function filterFriends() {
+		clearTimeout(friendsTimeout);
+		var nw = $('.invite-filters').attr('data-nw-selected');
+		var filter = $('.invite-filter').val();
+		friendsTimeout = setTimeout(function () { prepInviteTab(filter, nw) }, 200);
```
Comment (2is10) [inline kifi-backend/modules/common/public/site/js/main.js:575]: consider:

```
-  function () { prepInviteTab(filter, nw) }
+  prepInviteTab.bind(null, filter, nw)
```

File: kifi-backend/modules/common/public/site/js/main.js [hunk 2]
```diff
@@ -566,29 +566,30 @@ $(function() {
 		nwFriendsScroller.refresh();
 	});
 	var $nwFriendsLoading = $('.invite-friends-loading');
-	function prepInviteTab() {
-		$.getJSON(xhrBase + '/user/socialConnections', function(friends) {
+	var friendsToShow = 20;
+	var friendsTimeout;
+	function filterFriends() {
+		clearTimeout(friendsTimeout);
+		var nw = $('.invite-filters').attr('data-nw-selected');
```
Comment (2is10) [inline kifi-backend/modules/common/public/site/js/main.js:573]: consider `.data('nwSelected')` although I understand if you prefer the longer form to facilitate code search

File: kifi-backend/modules/common/public/site/index.html [hunk 3]
```diff
@@ -285,7 +286,6 @@ <h3 class="collection" data-template data-id="{{id}}" style="display:none">
 							{% endif %}
 						</li>
 					</ul>
-					<img class="invite-friends-loading" src="img/wait.gif">
```
Comment (2is10) [inline kifi-backend/modules/common/public/site/index.html:288]: I think this was originally at the bottom to facilitate infinite scroll. Is infinite scroll a TODO? (when there is no name filter)

**Comment by 2is10** [comment]
Looks good!

File: kifi-backend/modules/common/public/site/index.html (hunk 3 above)
Comment (gmethvin) [inline kifi-backend/modules/common/public/site/index.html:288]: Infinite scroll doesn't seem to work for me. Did you implement it here?

Comment (gmethvin) [inline kifi-backend/modules/common/public/site/index.html:288]: I'm planning to add it for the no name filter case. Alternatively, we could scrap infinite scroll altogether and require a search query. Seems like a product question though.

Comment (2is10) [inline kifi-backend/modules/common/public/site/index.html:288]: Cool. No, I didn't implement it... I was just loading all connections in one go. I think I saw an Asana ticket for it.

**Pull request merged**